src/main/java/cz/brmlab/yodaqa/flow/dashboard/SourceIDGenerator.java

```diff
@@ -0,0 +1,23 @@
+package cz.brmlab.yodaqa.flow.dashboard;
+
+/**
+ * Created by nhl on 7/9/15.
+ */
+
+
+/**
+ * This class should generate unique ID's for sources!
+ * The ID's are unique for the session
+ * It is thread-safe
+ */
+public final class SourceIDGenerator {
+	/*Singleton*/
+	volatile int IDCounter = 0;
```

Comment:
Yeah, it is probably more for the cases when we need to create objects and access them at the same time, I'll change this to private. Now that I'm thinking about it, volatile int was actually correct, because the counter wasn't private...